nominal/core.py

```diff
@@ -709,6 +866,35 @@ def create_video_from_io(
         response = self._ingest_client.ingest_video(self._auth_header, request)
         return self.get_video(response.video_rid)
 
+    def create_log_set(
+        self,
+        name: str,
+        timestamp_type: TimestampType,
+        description: str | None = None,
+        origin_metadata: Mapping[str, str] = MappingProxyType({}),
+    ) -> LogSet:
+        """
+        Creates a log set, to which logs can be attached using `attach-and-finalize`. The logs within a logset are
+        not searchable until the logset is finalized.
+        Timestamp type must be a string equal to either 'ABSOLUTE' or 'RELATIVE'.
+        """
+        timestamp_type_enum = None
+        if timestamp_type == "absolute":
+            timestamp_type_enum = datasource.TimestampType.ABSOLUTE
+        elif timestamp_type_enum == "relative":
+            timestamp_type_enum = datasource.TimestampType.RELATIVE
+        else:
+            raise TypeError(f"timestamp type {timestamp_type} must be one of [RELATIVE, ABSOLUTE]")
```

Comment:
suggestion: if you can ever get into the situation where you have to initialize a variable because branches may risk it not being defined, break it out into a function. This ensures you'll never end up in a branch with an undefined variable and makes type checking easy and explicit. E.g.:

```py
def convert_timestamp(t: Literal["absolute", "relative"]) -> datasource.TimestampType:
    if t == "absolute":
        return datasource.TimestampType.ABSOLUTE
    if t == "relative":
        return = datasource.TimestampType.RELATIVE
    raise ValueError(f"timestamp type {t} must be one of [RELATIVE, ABSOLUTE]")
```

now the caller will _always_ have a value of type `datasource.TimestampType`.

Reply:
also unrelated nit, but ValueError is more semantic here - the value is of the correct type (str), just an incorrect value.